modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java

```diff
@@ -403,12 +403,24 @@ public void importUnencryptedDatabase(final String accountData, final String pas
     @ReactMethod
     public void logout() {
         Log.d(TAG, "logout");
-        String result = Statusgo.logout();
-        if (result.startsWith("{\"error\":\"\"")) {
-            Log.d(TAG, "Logout result: " + result);
-        } else {
-            Log.e(TAG, "Logout failed: " + result);
+        if (!checkAvailability()) {
+            System.exit(0);
+            return;
         }
+
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                String result = Statusgo.logout();
+                if (result.startsWith("{\"error\":\"\"")) {
+                    Log.d(TAG, "Logout result: " + result);
+                } else {
+                    Log.e(TAG, "Logout failed: " + result);
+                }
+            }
+        };
+
+        StatusThreadPoolExecutor.getInstance().execute(r);
```

Comment:
Even when the issue is fixed, `Statusgo.logout` sometime takes 2-3 seconds to return the result. So it's better to use `StatusThreadPoolExecutor` so that UI doesn't look frozen. (this will show the progress bar instead)